DatadogInternal/Sources/NetworkInstrumentation/Datadog/HTTPHeadersWriter.swift

```diff
@@ -58,7 +58,7 @@ public class HTTPHeadersWriter: TracePropagationHeadersWriter {
     /// Initializes the headers writer.
     ///
     /// - Parameter sampler: The sampler used for headers injection.
-    public init(sampler: Sampler) {
+    public init(sampler: Sampling) {
```

Comment:
❗ 
This is a breaking change

Reply:
Technically yes, but the `Sampler` type is only defined in `DatadogInternal` without being re-exported from user-facing modules. In practise, the only supported public initializer for `HTTPHeadersWriter` is `init(sampleRate:)`, which is not altered by this change.